pkg/policy/directory/watcher.go

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package directory
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+
+	ipcacheTypes "github.com/cilium/cilium/pkg/ipcache/types"
+	k8sConst "github.com/cilium/cilium/pkg/k8s/apis/cilium.io"
+	k8sCiliumUtils "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/utils"
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/policy"
+	"github.com/cilium/cilium/pkg/source"
+)
+
+type policyWatcher struct {
+	log           logrus.FieldLogger
+	config        *option.DaemonConfig
+	policyManager PolicyManager
+	// maps cnp file name to cnp object. this is required to retrieve data during delete.
+	fileNameToCnpCache map[string]*cilium_v2.CiliumNetworkPolicy
+}
+
+func (p *policyWatcher) translateToCNPObject(file string) (*cilium_v2.CiliumNetworkPolicy, error) {
+	data, err := os.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	// yaml to json conversion
+	jsonData, err := yaml.YAMLToJSON([]byte(data))
+	if err != nil {
+		return nil, err
+	}
+
+	// translate json to cnp object
+	cnp := &cilium_v2.CiliumNetworkPolicy{}
+	err = json.Unmarshal(jsonData, cnp)
+	if err != nil {
+		return nil, err
+	}
+
+	return cnp, nil
+}
+
+func getLabels(filePath string, cnp *cilium_v2.CiliumNetworkPolicy) labels.LabelArray {
+	name := cnp.ObjectMeta.Name
+	ns := cnp.ObjectMeta.Namespace
+
+	labelsArr := labels.LabelArray{
+		labels.NewLabel("name", name, labels.LabelSourceDirectory),
+		labels.NewLabel("path", filePath, labels.LabelSourceDirectory),
```

Comment:
Kubernetes has restrictions on the characters that can be put into labels, as well as the label's format (see [here](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set)). The `filePath` argument needs to undergo input validation to ensure it fits within these guidelines. For example, the forward-slashes in the path should probably be replaced with a different character, such as an underscore, since the forward-slash character has a specific meaning. Additionally, Kubernetes labels are restricted to ASCII characters, but file paths can contain UTF-8 characters.

Reply:
Ah interesting that's good to know, TIL. Do we have the format of Cilium labels documented somewhere?

Reply:
instead of absolute path, i'm planning to add only name of file. 
1. Anyway name of file will be unique under directory and directory watcher listens only on one directory
2. can avoid longer filepaths